vertx-web-client/src/test/java/io/vertx/ext/web/client/it/UriTemplateITest.java

```diff
@@ -0,0 +1,90 @@
+package io.vertx.ext.web.client.it;
+
+
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpVersion;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static io.vertx.core.Vertx.vertx;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(VertxUnitRunner.class)
+public class UriTemplateITest {
+
+  private HttpClient client = vertx().createHttpClient();
+
+  private Vertx vertx;
+
+  @Before
+  public void deploy(TestContext context) {
+    vertx = Vertx.vertx();
+    Async async = context.async();
+    vertx.deployVerticle(UriTemplateVerticle.class.getName());
+    async.complete();
+  }
+
+  @Test
+  public void greetingFromUriTemplateTest(TestContext testContext) {
+    client.request(HttpMethod.GET, 8089, "localhost", "/greeting")
+      .compose(HttpClientRequest::send)
+      .onComplete(testContext.asyncAssertSuccess(resp -> {
+        testContext.assertEquals(200, resp.statusCode());
+        testContext.assertEquals(HttpVersion.HTTP_1_1, resp.version());
+        resp.body(bufferAsyncResult -> {
+          if (bufferAsyncResult.succeeded()) {
+            testContext.assertEquals("Hello from UriTemplateVerticle!", bufferAsyncResult.result().toString());
+          } else {
+            System.out.println(bufferAsyncResult.cause().getMessage());
+          }
+        });
+
+      })).onFailure(throwable -> throwable.getMessage());
```

Comment:
no need for onFailure here, onComplete(testContext.asyncAssertSuccess(...)) is enough